### PR TITLE
[parsec] Improve typing

### PR DESCRIPTION
**Pull request by FirelightFlagboy**
- Replace `BaseWorkspaceStorage` with `AnyWorkspaceStorage`
    > This change will make more sense when we merge #2903 
- Replace `Tuple` by `tuple`, `Dict` by `dict`, etc
- Move import mostly to `parsec._parsec`

### CODE REVIEW

File: parsec/core/fs/workspacefs/workspacefs.py [hunk 1]
```diff
@@ -4,16 +4,32 @@
...
+    LocalFolderManifest,
+    LocalWorkspaceManifest,
+    MaintenanceType,
+    RealmID,
+    RealmStatusRepOk,
+    Regex,
+    UserID,
+    WorkspaceEntry,
+)
+from parsec._parsec import FileManifest as RemoteFileManifest
```
Comment (Max-7) [inline parsec/core/fs/workspacefs/workspacefs.py:31]: Dupplicate. I think you can use the alias in the import tuple, eg
```
from parsec._parsec import (
    EntryID,
    EntryName,
    ...
    FileManifest as RemoteFileManifest
)
```

File: parsec/core/fs/workspacefs/workspacefs.py [hunk 2]
```diff
@@ -252,7 +261,9 @@ def is_read_only(self) -> bool:
     def is_revoked(self) -> bool:
         return self.get_workspace_entry().role is None
 
-    async def path_info(self, path: AnyPath) -> Dict[str, object]:
+    # Information
```
Comment (Max-7) [inline parsec/core/fs/workspacefs/workspacefs.py:264]: Useless comment, the function name is explicit enough.

File: parsec/core/fs/workspacefs/workspacefs.py [hunk 3]
```diff
@@ -4,16 +4,32 @@
...
-from parsec.api.data import AnyRemoteManifest, BlockAccess
-from parsec.api.data import FileManifest as RemoteFileManifest
-from parsec.api.protocol import MaintenanceType, RealmID, UserID
+from parsec._parsec import (
+    BlockAccess,
+    CoreEvent,
+    DateTime,
+    EntryID,
+    EntryName,
+    FileManifest,
```
Comment (Max-7) [inline parsec/core/fs/workspacefs/workspacefs.py:19]: `FileManifest` was previously imported from `parsec.api.data` as `RemoteFileManifest`, and from `parsec._parsec` as `FileManifest`. What's the difference between these two? Is the alias useless? Do we have two different classes with the same name?

File: parsec/core/fs/workspacefs/file_operations.py [hunk 4]
```diff
@@ -2,7 +2,18 @@
...
+    Chunk,
+    ChunkID,
+    prepare_read,
+    prepare_reshape,
+    prepare_resize,
+    prepare_write,
+)
+
+Chunks = tuple[Chunk, ...]
+ChunkIDSet = set[ChunkID]
```
Comment (Max-7) [inline parsec/core/fs/workspacefs/file_operations.py:15]: Are those used somewhere? I'm not a big fan of the alias on simple things like lists, it makes it less explicit. If we take something like `Chunks` and I see this type in the codebase, now I have to wonder weither it's a class on its own or a collection, and what type of collection.

**Review by Max-7 — CHANGES_REQUESTED** [review]

File: parsec/core/fs/workspacefs/workspacefs.py (hunk 1 above)
Comment (FirelightFlagboy) [reply parsec/core/fs/workspacefs/workspacefs.py:31]: Ruff force to have a new line when we write an alias :/

File: parsec/core/fs/workspacefs/workspacefs.py (hunk 3 above)
Comment (FirelightFlagboy) [reply parsec/core/fs/workspacefs/workspacefs.py:19]: `FileManifest` from `parsec.api.data` is a re-export of `parsec._parsec`

File: parsec/core/fs/workspacefs/workspacefs.py (hunk 2 above)
Comment (FirelightFlagboy) [reply parsec/core/fs/workspacefs/workspacefs.py:264]: done

File: parsec/core/fs/workspacefs/file_operations.py (hunk 4 above)
Comment (FirelightFlagboy) [reply parsec/core/fs/workspacefs/file_operations.py:15]: Those alias aren't used, I've removed them

**Review by Max-7 — APPROVED** [review]

**Pull request merged**